### ISSUE TITLE
clang-format

### ISSUE THREAD

**Issue by Thunder07** [jpd002/Play-#686]
I ran `clang-format` on this repo and I noticed that you're using `UseTab: ForIndentation` that results in alignment using spaces (look for the comment in the code)
e.g
```C
	lastOpenCommand m_lastOpenCommand;
	QString         m_lastpath = QDir::homePath(); //<-- the space between QString and m_lastpath is made with space
```
```C
	QMessageBox::StandardButton resBtn = QMessageBox::question(this, "Close Confirmation?",
	                                                           tr("Are you sure you want to exit?\nHave you saved your progress?\n"),< ---all spaces
	                                                           QMessageBox::Yes | QMessageBox::No, <--
	                                                           QMessageBox::Yes);<---
```
```C
	for(auto listenerIterator(std::begin(m_listeners));
	    listenerIterator != std::end(m_listeners); listenerIterator++)// <--- this line starts with tab, then is aligned with spaces
	{
```
this can be changed by setting `UseTab: Always`

and I have a question, how do you want to handle CI clang format, I was thinking a separate build where only `clang-format` is called, if it finds changes, get it to generate commit.patch (print it to terminal or upload it to server, you can use travis hooks, to trigger a bot to post a link of that patch to GitHub so it can be added to the PR, if it was in a PR that is) and cause the build to fail so you'd notice it.

**Comment by jpd002** [comment]
Yup, that's right, I meant to use tabs for indent and spaces for alignment. Most of the previous code is using tabs for alignment, but I started using spaces recently.

I think a clang-format build should only check that all code is formatted correctly (ie.: that no change is generated when clang-format is applied on every source file.) I feel like automating a patch generation and upload process is going to be too complicated. I think it's not too much to ask to run clang-format before submitting a PR.

**Comment by Thunder07** [comment]
alright, I've been playing with `clang-format` and found that different versions produce different result due so some new options getting removed or new ones added in, which means to be able to have a coherent testing platform using travis, you'd either need to force everyone to use the same version of clang, or use travis to push out clang-format patch which then either you or the user that open the PR can attach to it. realistically i think the latter option is better, so quick google search i found this http://termbin.com/ which we can use to push the patches to.

so, which choice would you go for, and what version of clang-format are you using.
I know on my desktop the default is 4.0.1 and has issues finding `BreakConstructorInitializers: BeforeComma`, while the default travis one is 5.0 but we can install anything from 3.4->5.0 on it.

or you can just run clang-format once, and be rigorous with the PR submissions going forward.

**Comment by Thunder07** [comment]
well the defualt one is v5.0 but according to [this](https://github.com/juchem/apt-source-whitelist/blob/master/ubuntu.json#L238) we should be able to add v6.0 to travis with just a single extra line.

ya, my original train of thoughts was to use notification to get a bot to post it into the PR, while a bit overboard It would be organised, while using termbin is much more simplified process.

**Comment by jpd002** [comment]
PR has been merged (#689). Closing issue.
